presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -203,7 +206,14 @@ private static ConnectorPageSource createSelectivePageSource(
             }
         }
 
-        throw new IllegalStateException("Could not find a file reader for split " + split);
+        String errorMessage = format("Cannot read file in format %s: %s. " +
+                "Optimized table scan supports only ORC and DWRF files. " +
+                "It is disabled for tables with schemas other than these specified in the Metastore. " +
```

Comment:
I think this expresses the situation in a confusing way.

Is schemas the correct way to refer to the file format? Also metastore doesn't support multiple formats per table so it would be "other than what is specified in Metastore" (also "the" is optional).